core_backend/migrations/versions/2024_07_18_e354e0c0aad1_add_generate_llm_response_column.py

```diff
@@ -0,0 +1,42 @@
+"""add generate_llm_response column to query table
+
+Revision ID: e354e0c0aad1
+Revises: c6258d82bfd5
+Create Date: 2024-07-18 02:26:05.749460
+
+"""
+
+from typing import Sequence, Union
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision: str = "e354e0c0aad1"
+down_revision: Union[str, None] = "c6258d82bfd5"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column(
+        "query", sa.Column("query_generate_llm_response", sa.BOOLEAN(), nullable=True)
+    )
+    # WARNING: The query_generate_llm_response column for ALL previous queries will be
+    # set to False even if the user had requested an LLM response (since we don't
+    # collect this info yet).
```

Comment:
Is the alternative to look through query responses to see which ones had "llm_response" or not? But even then if a guardrail fails, say at identify_language step, I guess there is no way to know whether the request was to /embeddings-search or /llm-response?

Either way, backfilling with False shouldn't be a problem!